### ISSUE TITLE
Using codecs.load_all_yaml with generic resources

### ISSUE THREAD

**Issue by stonepreston** [gtsystem/lightkube#18]
I added the following generic resource for a CRD that is present in my cluster:
```python
SparkApp = create_namespaced_resource(group="sparkoperator.k8s.io", version="v1beta2", kind="SparkApplication", plural="sparkapplications")
```
and then attempted to create an object by loading from this [yaml file](https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/blob/master/examples/spark-pi.yaml) using:
```python
with open('../../examples/spark-pi.yaml') as f:
    for obj in codecs.load_all_yaml(f):
        client.create(obj, namespace="spark")
```
But I get an error:
```
lightkube.core.exceptions.LoadResourceError: No module named 'lightkube.resources.sparkoperator_v1beta2'
```
However I can create it this way by reading the yaml into a dict, creating a SparkApp object using that dict, and then using client.create:
```python
with open('../../examples/spark-pi.yaml', 'r') as stream:
    try:
        parsed_yaml = yaml.safe_load(stream)
        app = SparkApp(parsed_yaml)
        client.create(app, namespace="spark")
    except yaml.YAMLError as exc:
        print(exc)
```
Should I be able to create this using load_all_yaml? It seems as if load_all_yaml is not aware of my generic resource.

**Comment by gtsystem** [comment]
In general it should work, but since your custom resource group ends with ".k8s.io" the load functions assumes this is part of the standard k8s resources and skip the crd check. See code [here](https://github.com/gtsystem/lightkube/blob/master/lightkube/codecs.py#L22). I guess this assumption is wrong and we will need to always check crd first.

**Comment by stonepreston** [comment]
> In general it should work, but since your custom resource group ends with ".k8s.io" the load function assumes this is part of the standard k8s resources and skips the crd check. See code [here](https://github.com/gtsystem/lightkube/blob/master/lightkube/codecs.py#L22). I guess this assumption is wrong and we will need to always check crd first.

Awesome if I find some time in the next day or 2 I may be able to contribute a fix + test for this.

**Comment by stonepreston** [comment]
Closed with PR #21
